cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -248,99 +248,111 @@ def upload_rna_coverage_bigwig_to_scout(self, case_id: str, dry_run: bool) -> No
+                rna_dna_sample_case_map=rna_dna_sample_case_map, is_splice_junctions=False
+            )
+        )
         LOG.info("Upload RNA coverage bigwig file finished!")
 
     def upload_splice_junctions_bed_to_scout(self, dry_run: bool, case_id: str) -> None:
         """Upload splice_junctions_bed file for a case to Scout."""
 
         status_db: Store = self.status_db
-        rna_case: Family = status_db.get_case_by_internal_id(internal_id=case_id)
 
         rna_dna_sample_case_map: Dict[str, Dict[str, list]] = self.create_rna_dna_sample_case_map(
-            rna_case=rna_case
+            rna_case=status_db.get_case_by_internal_id(internal_id=case_id)
         )
         for rna_sample_id in rna_dna_sample_case_map:
             splice_junctions_bed: Optional[File] = self.get_splice_junctions_bed(
                 case_id=case_id, sample_id=rna_sample_id
             )
-
             if splice_junctions_bed is None:
                 raise FileNotFoundError(
                     f"No splice junctions bed file was found in Housekeeper for {rna_sample_id}"
                 )
 
             LOG.info(f"Splice junctions bed file {splice_junctions_bed.path} found")
-            dna_sample_id: str
-            dna_cases: List[str]
-            dna_sample_id, dna_cases = rna_dna_sample_case_map[rna_sample_id].popitem()
-            for dna_case_id in dna_cases:
-                if self.status_db.get_case_by_internal_id(internal_id=dna_case_id).is_uploaded:
-                    LOG.info(
-                        f"Uploading splice junctions bed file for sample {dna_sample_id} in case {dna_case_id} in Scout."
-                    )
+            self._upload_splice_junctions_bed_or_coverage_for_rna_sample(
+                file=splice_junctions_bed,
+                rna_sample_id=rna_sample_id,
+                rna_dna_sample_case_map=rna_dna_sample_case_map,
+                dry_run=dry_run,
+                is_splice_junctions=True,
+            )
 
-                    if dry_run:
-                        continue
+        LOG.info(
+            self.summary_for_rna_case_splice_junctions_or_coverage_upload(
+                rna_dna_sample_case_map=rna_dna_sample_case_map, is_splice_junctions=True
+            )
+        )
+        LOG.info("Upload splice junctions bed file finished!")
 
+    def _upload_splice_junctions_bed_or_coverage_for_rna_sample(
+        self,
+        file: Optional[File],
+        rna_sample_id: str,
+        rna_dna_sample_case_map: Dict[str, Dict[str, list]],
+        dry_run: bool,
+        is_splice_junctions: bool,
+    ) -> None:
+        """Uploads splice junctions bed or bigwig coverage
+        for an RNA sample.
+        """
+
+        (dna_sample_id, dna_cases) = rna_dna_sample_case_map[rna_sample_id].popitem()
+        file_name: str = (
+            "splice junctions bed file" if is_splice_junctions else "bigwig coverage file"
+        )
```

Comment:
Maybe is worth doing a function for this operation as it is invoked twice (that I have noticed) to compile with DRY

Reply:
Good suggestion, done!